WalletWasabi.Gui/Global.cs

```diff
@@ -172,10 +172,19 @@ public async Task InitializeNoWalletAsync(TerminateService terminateService)
 
 				#region BitcoinStoreInitialization
 
-				await bstoreInitTask.ConfigureAwait(false);
+				try
+				{
+					await bstoreInitTask.ConfigureAwait(false);
 
-				// Make sure that the height of the wallets will not be better than the current height of the filters.
-				WalletManager.SetMaxBestHeight(BitcoinStore.IndexStore.SmartHeaderChain.TipHeight);
+					// Make sure that the height of the wallets will not be better than the current height of the filters.
+					WalletManager.SetMaxBestHeight(BitcoinStore.IndexStore.SmartHeaderChain.TipHeight);
+				}
+				catch
+				{
+					// If our internal data structures in the Bitcoin Store gets corrupted, then it's better to rescan all the wallets.
+					WalletManager.SetMaxBestHeight(0);
```

Comment:
As meantioned in your [comment](https://github.com/zkSNACKs/WalletWasabi/pull/5496/files#r604811707), is it better to set it to the block height of the first bech32 filter (481824)?